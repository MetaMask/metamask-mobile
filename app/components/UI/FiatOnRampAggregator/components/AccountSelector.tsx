import React from 'react';

import { StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';

import EthereumAddress from '../../EthereumAddress';
import JSIdenticon from '../../Identicon';
import Text from '../../../Base/Text';
import JSSelectorButton from '../../../Base/SelectorButton';
import { useNavigation } from '@react-navigation/native';
import { createAccountSelectorNavDetails } from '../../../Views/AccountSelector';

// TODO: Convert into typescript and correctly type
const SelectorButton = JSSelectorButton as any;
const Identicon = JSIdenticon as any;

const styles = StyleSheet.create({
  selector: {
    flexShrink: 1,
  },
  accountText: {
    flexShrink: 1,
    marginVertical: 3,
    marginHorizontal: 5,
  },
});

const AccountSelector = () => {
  const navigation = useNavigation();
  const selectedAddress = useSelector(
    (state: any) =>
      state.engine.backgroundState.PreferencesController.selectedAddress,
  );

  const identities = useSelector(
    (state: any) =>
      state.engine.backgroundState.PreferencesController.identities,
  );

  const openAccountSelector = () =>
    navigation.navigate(...createAccountSelectorNavDetails());

  return (
    <SelectorButton onPress={openAccountSelector} style={styles.selector}>
      <Identicon diameter={15} address={selectedAddress} />
      <Text style={styles.accountText} primary centered numberOfLines={1}>
        {identities[selectedAddress]?.name} (
        <EthereumAddress address={selectedAddress} type={'short'} />)
      </Text>
    </SelectorButton>
  );
};

export default AccountSelector;
