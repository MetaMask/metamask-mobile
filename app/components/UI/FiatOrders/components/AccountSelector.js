import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';

import { toggleAccountsModal } from '../../../../actions/modals';
import EthereumAddress from '../../EthereumAddress';
import Identicon from '../../Identicon';
import Text from '../../../Base/Text';
import SelectorButton from '../../../Base/SelectorButton';

const styles = StyleSheet.create({
  selector: {
    flexShrink: 1,
  },
  accountText: {
    flexShrink: 1,
    marginVertical: 3,
    marginHorizontal: 5,
  },
});
const AccountSelector = ({
  toggleAccountsModal,
  selectedAddress,
  identities,
}) => (
  <SelectorButton onPress={toggleAccountsModal} style={styles.selector}>
    <Identicon diameter={15} address={selectedAddress} />
    <Text style={styles.accountText} primary centered numberOfLines={1}>
      {identities[selectedAddress]?.name} (
      <EthereumAddress address={selectedAddress} type={'short'} />)
    </Text>
  </SelectorButton>
);

AccountSelector.propTypes = {
  toggleAccountsModal: PropTypes.func.isRequired,
  selectedAddress: PropTypes.string.isRequired,
  identities: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  selectedAddress:
    state.engine.backgroundState.PreferencesController.selectedAddress,
  identities: state.engine.backgroundState.PreferencesController.identities,
});

const mapDispatchToProps = (dispatch) => ({
  toggleAccountsModal: () => dispatch(toggleAccountsModal()),
});
export default connect(mapStateToProps, mapDispatchToProps)(AccountSelector);
