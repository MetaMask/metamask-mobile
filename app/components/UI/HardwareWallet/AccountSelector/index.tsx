import React, { useCallback, useState, useMemo } from 'react';
import { View, Text, FlatList, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import CheckBox from '@react-native-community/checkbox';
import { useSelector } from 'react-redux';

import { strings } from '../../../../../locales/i18n';
import { IAccount } from './types';
import useBlockExplorer from '../../../hooks/useBlockExplorer';
import { useAccountsBalance } from './hooks';
import { useTheme } from '../../../../util/theme';
import { createStyle } from './styles';
import AccountDetails from '../AccountDetails';
import StyledButton from '../../../UI/StyledButton';
import { selectProviderConfig } from '../../../../selectors/networkController';

interface ISelectQRAccountsProps {
  accounts: IAccount[];
  selectedAccounts: string[];
  nextPage: () => void;
  prevPage: () => void;
  toggleAccount: (index: number) => void;
  onUnlock: (accountIndex: number[]) => void;
  onForget: () => void;
  title: string;
}

const AccountSelector = (props: ISelectQRAccountsProps) => {
  const {
    accounts,
    prevPage,
    nextPage,
    toggleAccount,
    selectedAccounts,
    onForget,
    onUnlock,
    title,
  } = props;

  const { colors } = useTheme();
  const styles = createStyle(colors);
  const providerConfig = useSelector(selectProviderConfig);
  const accountsBalance = useAccountsBalance(accounts);
  const { toBlockExplorer } = useBlockExplorer();

  const [checkedAccounts, setCheckedAccounts] = useState<Set<number>>(
    new Set(),
  );

  const formattedAccounts: IAccount[] = useMemo(() => {
    const selectedAccountsSet = new Set<string>(
      selectedAccounts.map((address) => address.toLowerCase()),
    );
    return accounts.map((account) => {
      const checked = checkedAccounts.has(account.index);
      const selected = selectedAccountsSet.has(account.address.toLowerCase());
      return {
        ...account,
        checked: checked || selected,
        exist: selected,
        balance: accountsBalance[account.address]?.balance || '0x0',
      };
    });
  }, [accounts, checkedAccounts, selectedAccounts, accountsBalance]);

  const onCheckBoxClick = useCallback(
    (index: number) => {
      setCheckedAccounts((prev: Set<number>) => {
        prev.has(index) ? prev.delete(index) : prev.add(index);
        return new Set(prev);
      });
      toggleAccount(index);
    },
    [toggleAccount],
  );

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <FlatList
        data={formattedAccounts}
        keyExtractor={(item) => `address-${item.index}`}
        renderItem={({ item }) => (
          <View style={[styles.account]}>
            <CheckBox
              style={[styles.checkBox]}
              disabled={item.exist}
              value={item.checked}
              onValueChange={() => onCheckBoxClick(item.index)}
              boxType={'square'}
              tintColors={{
                true: colors.primary.default,
                false: colors.border.default,
              }}
              onCheckColor={colors.background.default}
              onFillColor={colors.primary.default}
              onTintColor={colors.primary.default}
            />
            <AccountDetails
              index={item.index}
              address={item.address}
              balance={item.balance}
              ticker={providerConfig.ticker}
              toBlockExplorer={toBlockExplorer}
            />
          </View>
        )}
      />
      <View style={styles.pagination}>
        <TouchableOpacity style={styles.paginationItem} onPress={prevPage}>
          <Icon name={'chevron-left'} color={colors.primary.default} />
          <Text style={styles.paginationText}>
            {strings('account_selector.prev')}
          </Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.paginationItem} onPress={nextPage}>
          <Text style={styles.paginationText}>
            {strings('account_selector.next')}
          </Text>
          <Icon name={'chevron-right'} color={colors.primary.default} />
        </TouchableOpacity>
      </View>
      <View style={styles.bottom}>
        <StyledButton
          type={'confirm'}
          onPress={() => onUnlock([...checkedAccounts])}
          containerStyle={[styles.button]}
          disabled={checkedAccounts.size < 1}
        >
          {strings('account_selector.unlock')}
        </StyledButton>
        <StyledButton
          type={'transparent-blue'}
          onPress={onForget}
          containerStyle={[styles.button]}
        >
          {strings('account_selector.forget')}
        </StyledButton>
      </View>
    </View>
  );
};

export default AccountSelector;
