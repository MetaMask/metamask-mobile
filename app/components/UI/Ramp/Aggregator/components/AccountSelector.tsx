import React, { useCallback } from 'react';
import { StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import { isCaipChainId, toCaipChainId } from '@metamask/utils';

import SelectorButton from '../../../../Base/SelectorButton';
import Avatar, {
  AvatarSize,
  AvatarVariant,
} from '../../../../../component-library/components/Avatars/Avatar';
import Text, {
  TextVariant,
} from '../../../../../component-library/components/Texts/Text';

import { useAccountGroupName } from '../../../../hooks/multichainAccounts/useAccountGroupName';
import { selectSelectedInternalAccountFormattedAddress } from '../../../../../selectors/accountsController';
import { formatAddress } from '../../../../../util/address';
import { BuildQuoteSelectors } from '../../../../../../e2e/selectors/Ramps/BuildQuote.selectors';
import { createAddressSelectorNavDetails } from '../../../../Views/AddressSelector/AddressSelector';
import { getRampNetworks } from '../../../../../reducers/fiatOrders';
import { getNetworkImageSource } from '../../../../../util/networks';
import { selectChainId } from '../../../../../selectors/networkController';

const styles = StyleSheet.create({
  selector: {
    flexShrink: 1,
  },
  accountText: {
    flexShrink: 1,
    marginVertical: 3,
    marginHorizontal: 5,
  },
});

const AccountSelector = ({ isEvmOnly }: { isEvmOnly?: boolean }) => {
  const navigation = useNavigation();
  const selectedAddress = useSelector(
    selectSelectedInternalAccountFormattedAddress,
  );
  const selectedChainId = useSelector(selectChainId);
  const accountName = useAccountGroupName();

  const selectedFormattedAddress = useSelector(
    selectSelectedInternalAccountFormattedAddress,
  );

  const rampNetworks = useSelector(getRampNetworks);

  const selectedCaipChainId = isCaipChainId(selectedChainId)
    ? selectedChainId
    : toCaipChainId('eip155', selectedChainId);

  const rampNetworksCaipIds = rampNetworks.map((network) => {
    if (isCaipChainId(network.chainId)) {
      return network.chainId;
    }
    return toCaipChainId('eip155', network.chainId);
  });

  const openAccountSelector = useCallback(
    () =>
      navigation.navigate(
        ...createAddressSelectorNavDetails({
          isEvmOnly,
          displayOnlyCaipChainIds: rampNetworksCaipIds,
        }),
      ),
    [isEvmOnly, navigation, rampNetworksCaipIds],
  );

  const shortenedAddress = formatAddress(
    selectedFormattedAddress || '',
    'short',
  );

  const displayedAddress = accountName
    ? `(${shortenedAddress})`
    : shortenedAddress;

  return (
    <SelectorButton
      onPress={openAccountSelector}
      style={styles.selector}
      testID={BuildQuoteSelectors.ACCOUNT_PICKER}
    >
      {selectedAddress && selectedFormattedAddress ? (
        <>
          <Avatar
            variant={AvatarVariant.Network}
            size={AvatarSize.Xs}
            imageSource={getNetworkImageSource({
              chainId: selectedCaipChainId,
            })}
          />
          <Text
            variant={TextVariant.BodyMDMedium}
            style={styles.accountText}
            numberOfLines={1}
            ellipsizeMode="middle"
          >
            {accountName} {displayedAddress}
          </Text>
        </>
      ) : (
        <Text variant={TextVariant.BodyMD}>Account is loading...</Text>
      )}
    </SelectorButton>
  );
};

export default AccountSelector;
