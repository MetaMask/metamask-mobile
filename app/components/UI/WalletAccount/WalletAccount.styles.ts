import { StyleSheet } from 'react-native';
// External dependencies.
import { Theme } from '../../../util/theme/models';
import { WalletAccountStyleSheetVars } from './WalletAccount.types';

const styleSheet = (params: {
  theme: Theme;
  vars: WalletAccountStyleSheetVars;
}) => {
  const { vars, theme } = params;
  const { colors } = theme;
  const { style } = vars;
  return StyleSheet.create({
    base: {
      marginHorizontal: 16,
      borderWidth: 1,
      borderColor: colors.border.default,
      borderRadius: 8,
      ...style,
    },
    account: {
      alignItems: 'center',
    },
    accountPicker: {
      borderWidth: 0,
      borderColor: colors.border.default,
      borderRadius: 8,
    },
    middleBorder: {
      borderTopWidth: 1,
      borderColor: colors.border.default,
      marginHorizontal: 16,
    },
    addressContainer: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      backgroundColor: colors.background.default,
      padding: 20,
      borderWidth: 0,
      borderColor: colors.border.default,
      borderRadius: 8,
    },
  });
};
export default styleSheet;
