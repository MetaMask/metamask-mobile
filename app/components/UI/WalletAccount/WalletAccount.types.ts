import { ViewStyle } from 'react-native';

export interface WalletAccountProps {
  style?: ViewStyle;
}

/**
 * Style sheet input parameters.
 */
export type WalletAccountStyleSheetVars = Pick<WalletAccountProps, 'style'>;
