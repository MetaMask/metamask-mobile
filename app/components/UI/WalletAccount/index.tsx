/* eslint-disable import/prefer-default-export */
export { default as WalletAccount } from './WalletAccount';
