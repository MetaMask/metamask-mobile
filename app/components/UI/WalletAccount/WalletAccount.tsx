// Third parties dependencies
import React, {
  forwardRef,
  useCallback,
  useEffect,
  useImperativeHandle,
  useRef,
  useState,
} from 'react';
import { useSelector } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import { Platform, View } from 'react-native';
// External dependencies
import { IconName } from '../../../component-library/components/Icons/Icon';
import PickerAccount from '../../../component-library/components/Pickers/PickerAccount';
import { AvatarAccountType } from '../../../component-library/components/Avatars/Avatar/variants/AvatarAccount';
import { createAccountSelectorNavDetails } from '../../../components/Views/AccountSelector';
import { useStyles } from '../../../component-library/hooks';
import generateTestId from '../../../../wdio/utils/generateTestId';
import AddressCopy from '../AddressCopy';
import {
  doENSReverseLookup,
  isDefaultAccountName,
} from '../../../util/ENSUtils';
import { selectChainId } from '../../../selectors/networkController';
import ButtonIcon from '../../../component-library/components/Buttons/ButtonIcon/ButtonIcon';
import { ButtonIconSizes } from '../../../component-library/components/Buttons/ButtonIcon';
import Routes from '../../../constants/navigation/Routes';

// Internal dependencies
import styleSheet from './WalletAccount.styles';
import { WalletAccountProps } from './WalletAccount.types';
import {
  WALLET_ACCOUNT_ICON,
  MAIN_WALLET_ACCOUNT_ACTIONS,
} from '../../../../wdio/screen-objects/testIDs/Screens/WalletView.testIds';

const WalletAccount = ({ style }: WalletAccountProps, ref: React.Ref<any>) => {
  const { styles } = useStyles(styleSheet, { style });

  const { navigate } = useNavigation();
  const [ens, setEns] = useState<string>();

  const yourAccountRef = useRef(null);
  const accountActionsRef = useRef(null);

  useImperativeHandle(ref, () => ({
    yourAccountRef,
    accountActionsRef,
  }));
  /**
   * A string that represents the selected address
   */
  const selectedAddress = useSelector(
    (state: any) =>
      state.engine.backgroundState.PreferencesController.selectedAddress,
  );

  /**
   * An object containing each identity in the format address => account
   */
  const identities = useSelector(
    (state: any) =>
      state.engine.backgroundState.PreferencesController.identities,
  );

  const chainId = useSelector(selectChainId);

  const accountAvatarType = useSelector((state: any) =>
    state.settings.useBlockieIcon
      ? AvatarAccountType.Blockies
      : AvatarAccountType.JazzIcon,
  );
  const account = {
    address: selectedAddress,
    ...identities[selectedAddress],
  };

  const lookupEns = useCallback(async () => {
    try {
      const accountEns = await doENSReverseLookup(account.address, chainId);

      setEns(accountEns);
      // eslint-disable-next-line no-empty
    } catch {}
  }, [account.address, chainId]);

  useEffect(() => {
    lookupEns();
  }, [lookupEns]);

  const onNavigateToAccountActions = () => {
    navigate(Routes.MODAL.ROOT_MODAL_FLOW, {
      screen: Routes.SHEET.ACCOUNT_ACTIONS,
    });
  };

  return (
    <View style={styles.base} ref={yourAccountRef}>
      <PickerAccount
        accountAddress={account.address}
        accountName={
          isDefaultAccountName(account.name) && ens ? ens : account.name
        }
        accountAvatarType={accountAvatarType}
        onPress={() => {
          navigate(...createAccountSelectorNavDetails({}));
        }}
        showAddress={false}
        cellAccountContainerStyle={styles.account}
        style={styles.accountPicker}
        {...generateTestId(Platform, WALLET_ACCOUNT_ICON)}
      />
      <View style={styles.middleBorder} />
      <View style={styles.addressContainer} ref={accountActionsRef}>
        <AddressCopy formatAddressType="short" />
        <ButtonIcon
          iconName={IconName.MoreHorizontal}
          size={ButtonIconSizes.Sm}
          onPress={onNavigateToAccountActions}
          {...generateTestId(Platform, MAIN_WALLET_ACCOUNT_ACTIONS)}
        />
      </View>
    </View>
  );
};
export default forwardRef(WalletAccount);
