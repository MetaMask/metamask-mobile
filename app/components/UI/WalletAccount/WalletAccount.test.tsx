// Third party dependencies.
import React from 'react';
import { fireEvent } from '@testing-library/react-native';

// External dependencies
import renderWithProvider from '../../../util/test/renderWithProvider';
import ClipboardManager from '../../../core/ClipboardManager';
import { createAccountSelectorNavDetails } from '../../../components/Views/AccountSelector';

// Internal dependencies
import WalletAccount from './WalletAccount';

const initialState = {
  settings: {
    useBlockieIcon: false,
  },
  engine: {
    backgroundState: {
      NetworkController: {
        providerConfig: {
          type: 'mainnet',
          nickname: 'Ethereum mainnet',
          ticket: 'eth',
          chainId: '1',
        },
      },
      PreferencesController: {
        selectedAddress: '0x',
        identities: { '0x': { name: 'Account 1' } },
      },
    },
  },
};

jest.mock('../../../core/ClipboardManager');

const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualReactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...actualReactNavigation,
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: () => ({ top: 0, left: 0, right: 0, bottom: 0 }),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest
    .fn()
    .mockImplementation((callback) => callback(initialState)),
}));

describe('WalletAccount', () => {
  it('renders correctly', () => {
    const { toJSON } = renderWithProvider(<WalletAccount />, {
      state: initialState,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows the account address', () => {
    const { getByTestId } = renderWithProvider(<WalletAccount />, {
      state: initialState,
    });
    expect(getByTestId('wallet-account-address')).toBeDefined();
  });

  it('copies the account address to the clipboard when the copy button is pressed', async () => {
    const { getByTestId } = renderWithProvider(<WalletAccount />, {
      state: initialState,
    });

    fireEvent.press(getByTestId('wallet-account-copy-button'));
    expect(ClipboardManager.setString).toHaveBeenCalledTimes(1);
  });

  it('should navigate to the account selector screen on account press', () => {
    const { getByTestId } = renderWithProvider(<WalletAccount />, {
      state: initialState,
    });

    fireEvent.press(getByTestId('account-picker'));
    expect(mockNavigate).toHaveBeenCalledWith(
      ...createAccountSelectorNavDetails({}),
    );
  });
});
