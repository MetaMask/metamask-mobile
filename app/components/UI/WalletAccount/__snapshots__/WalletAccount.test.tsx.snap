// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletAccount renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#BBC0C5",
      "borderRadius": 8,
      "borderWidth": 1,
      "marginHorizontal": 16,
    }
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#BBC0C5",
        "borderRadius": 8,
        "borderWidth": 0,
        "flexDirection": "row",
        "padding": 16,
      }
    }
    testID="account-picker"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "height": 32,
            "marginRight": 16,
            "overflow": "hidden",
            "width": 32,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#1593F2",
                "borderRadius": 16,
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
        >
          <RNSVGSvgView
            bbHeight={32}
            bbWidth={32}
            focusable={false}
            height={32}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 32,
                  "width": 32,
                },
              ]
            }
            width={32}
          >
            <RNSVGGroup>
              <RNSVGRect
                fill={4280574177}
                height={32}
                matrix={
                  Array [
                    -0.14608302856241107,
                    0.9892723329629884,
                    -0.9892723329629884,
                    -0.14608302856241107,
                    37.75501596919891,
                    -2.82533949364276,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={32}
                x={0}
                y={0}
              />
              <RNSVGRect
                fill={4294395137}
                height={32}
                matrix={
                  Array [
                    -0.02094241988335673,
                    -0.9997806834748455,
                    0.9997806834748455,
                    -0.02094241988335673,
                    -15.024871507114028,
                    40.323921289217864,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={32}
                x={0}
                y={0}
              />
              <RNSVGRect
                fill={4294736640}
                height={32}
                matrix={
                  Array [
                    -0.458649554484315,
                    0.8886172326549487,
                    -0.8886172326549487,
                    -0.458649554484315,
                    28.4800145149268,
                    38.60053302995323,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={32}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#24272A",
              "fontFamily": "Euclid Circular B",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
          testID="account-label"
        >
          Account 1
        </Text>
      </View>
    </View>
    <SvgMock
      color="#6A737D"
      height={20}
      name="ArrowDown"
      style={
        Object {
          "height": 20,
          "marginLeft": 16,
          "width": 20,
        }
      }
      width={20}
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "borderColor": "#BBC0C5",
        "borderTopWidth": 1,
        "marginHorizontal": 16,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#BBC0C5",
        "borderRadius": 8,
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#24272A",
            "fontFamily": "Euclid Circular B",
            "fontSize": 12,
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 20,
          }
        }
      >
        Address
        :
      </Text>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="wallet-account-copy-button"
      >
        <View
          accessible={true}
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0376C919",
              "borderRadius": 20,
              "flexDirection": "row",
              "marginLeft": 8,
              "opacity": 1,
              "paddingHorizontal": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0376C9",
                "fontFamily": "Euclid Circular B",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 20,
              }
            }
            testID="wallet-account-address"
          >
            0x
          </Text>
          <SvgMock
            color="#0376C9"
            height={16}
            name="Copy"
            style={
              Object {
                "height": 16,
                "marginLeft": 4,
                "width": 16,
              }
            }
            width={16}
          />
        </View>
      </RNGestureHandlerButton>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
      testID="main-wallet-account-actions"
    >
      <SvgMock
        color="#24272A"
        height={16}
        name="MoreHorizontal"
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
        width={16}
      />
    </TouchableOpacity>
  </View>
</View>
`;
